airflow/models/dag_version.py

```diff
@@ -0,0 +1,169 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+import logging
+import random
+import string
+from typing import TYPE_CHECKING
+
+import uuid6
+from sqlalchemy import Column, ForeignKey, Integer, UniqueConstraint, func, select
+from sqlalchemy.orm import relationship
+from sqlalchemy_utils import UUIDType
+
+from airflow.models.base import Base, StringID
+from airflow.utils import timezone
+from airflow.utils.session import NEW_SESSION, provide_session
+from airflow.utils.sqlalchemy import UtcDateTime, with_row_locks
+
+if TYPE_CHECKING:
+    from sqlalchemy.orm import Session
+
+
+log = logging.getLogger(__name__)
+
+
+def _gen_random_str():
+    return "".join(random.choices(string.ascii_letters + string.digits, k=10))
+
+
+class DagVersion(Base):
+    """Model to track the versions of DAGs in the database."""
+
+    __tablename__ = "dag_version"
+    id = Column(UUIDType(binary=False), primary_key=True, default=uuid6.uuid7)
+    version_number = Column(Integer, nullable=False, default=1)
+    version_name = Column(StringID(), default=_gen_random_str, nullable=False)
+    dag_id = Column(StringID(), ForeignKey("dag.dag_id", ondelete="CASCADE"), nullable=False)
```

Comment:
don't all fks need to be define not inline so we can give a name to them?

Reply:
We have a naming convention that applies the naming automatically. We no longer need to specify a name for the ORM. We should for the migration files: https://github.com/apache/airflow/blob/dc91d07e298e5347ad77af25eb3533a13c71a883/airflow/models/base.py#L36

Fkeys: %(table_name)s_%(column_0_name)s_fkey => dag_version_dag_id_fkey. 

If we want to give a name different from the naming convention name, that's when we need to do the Fkey separately.